psi4/src/psi4/libmints/twobody.cc

```diff
@@ -112,6 +116,65 @@ TwoBodyAOInt::TwoBodyAOInt(const TwoBodyAOInt &rhs) : TwoBodyAOInt(rhs.integral_
 
 TwoBodyAOInt::~TwoBodyAOInt() {}
 
+void TwoBodyAOInt::update_density(const SharedMatrix &D) {
+    
+    timer_on("Density Screen");
+#pragma omp parallel for
+    for (int M = 0; M < nshell_; M++) {
+        for (int N = 0; N < nshell_; N++) {
+            int m_start = bs1_->shell(M).function_index();
+            int num_m = bs1_->shell(M).nfunction();
+
+            int n_start = bs1_->shell(N).function_index();
+            int num_n = bs1_->shell(N).nfunction();
+            
+            double max_dens = 0.0;
+
+            for (int m = m_start; m < m_start + num_m; m++) {
+                for (int n = n_start; n < n_start + num_n; n++) {
+                    double val = fabs(D->get(m, n));
+                    if (val > max_dens) max_dens = val;
+                }
+            }
+            max_dens_shell_pair_[M][N] = max_dens;
+        }
+    }
+    timer_off("Density Screen");
+
+}
+
+bool TwoBodyAOInt::shell_significant_density(int M, int N, int R, int S) {
+
+    Options& options = Process::environment.options;
+    double density_threshold = options.get_double("DENSITY_SCREENING_THRESHOLD");
+
+    double Q_MN = sqrt(shell_pair_values_[N * nshell_ + M]);
+    double Q_RS = sqrt(shell_pair_values_[S * nshell_ + R]);
```

Comment:
IMHO the square root should already be taken when `shell_pair_values` is formed; computing square roots for the screening may get expensive.

This would also simplify the screening code elsewhere, as one doesn't need the threshold for the squared quantity.